### DISCUSSION TITLE
Dask simulator

### DISCUSSION THREAD

**Discussion by erinaldi** [undark-lab/swyft#106] [Q&A]
Hello,

I would like to understand what the `DaskSimulator` is actually doing when `store.simulate()` is called and there is a `LocalCluster` object setup.
Right now I observe (in the Dark dashboard) that there is usually activity on all the workers are the beginning, but then most of the work is done by a single worker for 90% of the total time.
Is this the expected behavior? 

Thank you.

**Comment by bkmi** [comment]
@maxwelltsai @fnattino @meiertgrootes @rogerkuou  do you have any thoughts on this?

**Comment by fnattino** [comment]
Hi @erinaldi  - apologies for the very late reply, I completely missed the message (and @bkmi's ping!).

In case this is still relevant to you, the issue here could be the setting of `batch_size`, which regulates the size of the batches employed to run the simulations (see also #97 and #108). The default when calling `store.simulate()` is that all simulations are run in a single batch. This leads to all simulations being run sequentially on a single worker, which is probably what you have observed. Setting `batch_size` to ~ (total number of simulations) / (number of workers) or smaller should lead to the desired behaviour in which simulations are split and run across the various workers.

**Comment by erinaldi** [comment]
Hi @fnattino 

Thanks for the reply. Using `batch_size` might be indeed what is needed.
I will try it out.
